main.go

```diff
@@ -23,55 +33,51 @@ func main() {
 	statusCommand := flag.NewFlagSet("status", flag.ExitOnError)
 	statusToken := statusCommand.String("token", "", "stage the pipeline is currently in")
 
-	if len(os.Args) < 2 {
-		fmt.Println("new, promote, status or template subcommand is required")
-		os.Exit(1)
-		return
+	if len(args) < 2 {
+		return fmt.Errorf("new, promote or status subcommand is required")
 	}
 
 	path, err := setupFilesystem()
 	if err != nil {
-		fmt.Printf("ERROR: %v\n", err)
-		os.Exit(1)
-		return
+		return err
 	}
-	defer os.Remove(path)
+	defer func() { _ = os.Remove(path) }()
```

Comment:
If you are doing this to make the linter happy I suggest we handle the error.

Have a look at "#3" for an example of how to do this.
https://blog.learngoprogramming.com/5-gotchas-of-defer-in-go-golang-part-iii-36a1ab3d6ef1

Reply:
Ah, nice catch! Thanks :) Fixed, but skipped using named result.